packages/fuchsia_remote_debug_protocol/lib/src/dart/dart_vm.dart

```diff
@@ -0,0 +1,180 @@
+  Future<Map<String, dynamic>> invokeRpc(String function,
+      {Map<String, dynamic> params, Duration timeout}) async {
+    final Future<Map<String, dynamic>> future =
+        _peer.sendRequest(function, params ?? <String, dynamic>{});
+    if (timeout == null) {
+      return future;
+    }
+    return future.timeout(timeout, onTimeout: () {
+      throw new TimeoutException(
+          'Peer connection timed out during RPC call', timeout);
+    });
+  }
+
+  /// Returns a list of `FlutterView`s running across all Dart VM's.
+  ///
+  /// If there is no associated isolate with the flutter view (used to determine
+  /// the flutter view's name), then the flutter view's ID will be added
+  /// instead. If none of these things can be found (isolate has no name or the
+  /// flutter view has no ID), then the result will not be added to the list.
+  Future<List<FlutterView>> getAllFlutterViews() async {
+    final List<FlutterView> views = <FlutterView>[];
+    final Map<String, dynamic> rpcResponse =
+        await invokeRpc('_flutter.listViews', timeout: _kRpcTimeout);
+    final List<Map<String, dynamic>> flutterViewsJson = rpcResponse['views'];
+    for (Map<String, dynamic> jsonView in flutterViewsJson) {
+      final FlutterView flutterView = new FlutterView._fromJson(jsonView);
+      if (flutterView != null) {
+        views.add(flutterView);
+      }
+    }
+    return views;
+  }
+
+  /// Shuts down all active connections.
+  Future<Null> stop() async {
+    await _peer?.close();
+  }
+}
+
+/// Represents an instance of a Flutter view running on a Fuchsia device.
+class FlutterView {
+  /// Determines the name of the Isolate associated with this view. If there is
+  /// no associated Isolate, this will be set to the view's ID.
+  final String _name;
+
+  /// The ID of the Flutter view.
+  final String _id;
+
+  /// Attempts to construct a `FlutterView` from a json representation.
+  ///
+  /// If there is no Isolate and no id for the view, returns null. If there is
+  /// an associated isolate, and there is name for said isolate, also returns
+  /// null.
+  ///
+  /// All other cases return a `FlutterView` instance. The name of the
+  /// view may be null, but the id will always be set.
+  factory FlutterView._fromJson(Map<String, dynamic> json) {
+    final Map<String, dynamic> isolate = json['isolate'];
+    final String id = json['id'];
+    String name;
+    if (isolate != null) {
+      name = isolate['name'];
+      if (name == null) {
+        _log.warning('Unable to find name for isolate "$isolate"');
+        return null;
+      }
+    }
+
+    if (id == null) {
+      _log.warning('Unable to find view name for the following JSON structure '
+          '"$json"');
+      return null;
+    }
+
+    return new FlutterView._(name, id);
+  }
+
+  FlutterView._(this._name, this._id);
+
+  /// The ID of the `FlutterView`.
+  String get id => _id;
+
+  /// Returns the name of the `FucshiaFlutterView`.
```

Comment:
There is no more FucshiaFlutterView right?

Reply:
Woops! Looks like sed missed something.